noir-projects/aztec-nr/aztec/src/test/helpers/test_environment.nr

```diff
@@ -22,7 +22,7 @@ impl TestEnvironment {
     }
 
     pub unconstrained fn block_number(_self: Self) -> u32 {
-        get_block_number()
+        get_block_number() - 1
```

Comment:
This function needs to be documented. It's also just weird that we have a block_number() function and it returns different thing from get_block_number().

I think we should rename `TestEnvironment::block_number` to `TestEnvironment::committed_block_number` or smt. Or alternatively to historical_block_number since we call it like that [here](https://github.com/AztecProtocol/aztec-packages/blob/3e7134c7846f97029532333fccbf25612be1bd6c/noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/test.nr#L175).

Also do you know what block number would be used in private calls and what be used in public calls? Would it be the case that in private we would have output of env.block_number() and in public it would be incremented by 1?

Reply:
I agree. My intuition would be that `.block_number` would return the current block in which the tx is being "mined". 

Something like `previous_block_number` or `latest_committed_block_number` might work.

Presumably this function is only accessible in a Public context? In private, there is no notion of "the current block number" or "the previous block number", because it's computed async from the chain. Therefore, I think these functions should throw if called from a Private context.